src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs

```diff
@@ -620,6 +629,285 @@ public Task ChangeNickName(string nickName, WhoAmI whoAmI)
+                    {
+                        try
+                        {
+                            await StopFileTransfer(serverFileTransferId).ConfigureAwait(false);
+                        }
+                        catch
+                        { }
+
+                        throw new FileTransferException("File upload timed out.");
+                    }
+                }
+                else
+                {
+                    // Upload progress
+                    currentTimeoutMillis = 0;
+                    arrivedBytes = currentTransfer.SizeDone;
+                }
+
+                await Task.Delay(intervalMillis).ConfigureAwait(false);
+            }
+        }
+
+        #endregion
+
+        #region DownloadFile
+
+        public Task<byte[]> DownloadFile(int channelId, string filePath) => DownloadFile(channelId, string.Empty, filePath);
+
+        public async Task<byte[]> DownloadFile(int channelId, string channelPassword, string filePath)
+        {
+            var res = await Client.Send("ftinitdownload",
+                new Parameter("clientftfid", _fileTransferClient.GetFileTransferId()),
+                new Parameter("cid", channelId),
+                new Parameter("cpw", channelPassword),
+                new Parameter("name", NormalizePath(filePath)),
+                new Parameter("seekpos", 0)).ConfigureAwait(false);
+
+            var parsedRes = DataProxy.SerializeGeneric<InitDownload>(res).First();
+
+            if (parsedRes.Size > int.MaxValue)
+            {
+                throw new FileTransferException("The file is too big for a single byte array.");
+            }
+
+            return await _fileTransferClient.ReceiveFile((int)parsedRes.Size, parsedRes.Port, parsedRes.FileTransferKey).ConfigureAwait(false);
+        }
+
+        #endregion
+
+        #region GetCurrentFileTransfers
+
+        public async Task<IReadOnlyList<GetCurrentFileTransfer>> GetCurrentFileTransfers()
+        {
+            try
+            {
+                var res = await Client.Send("ftlist").ConfigureAwait(false);
+
+                return DataProxy.SerializeGeneric<GetCurrentFileTransfer>(res);
+            }
+            catch (QueryException ex)
+            {
+                if (ex.Error.Id == 1281)
+                {
+                    // For some reason this error occurs when there are no active file transfers
+                    return new ReadOnlyCollection<GetCurrentFileTransfer>(new GetCurrentFileTransfer[0]);
+                }
+                else
+                {
+                    throw ex;
+                }
+            }
+        }
+
+        #endregion
+
+        private Task StopFileTransfer(int serverFileTransferId, bool delete = true)
+        {
+            return Client.Send("ftstop",
+                new Parameter("serverftfid", serverFileTransferId),
+                new Parameter("delete", delete));
+        }
+
+        private string NormalizePath(string path)
```

Comment:
Is this path notation TeamSpeak-specific or can we re-use something from the standard library?

Reply:
The TeamSpeak3 server query requires paths to be in the following format:
- File: /path/to/file.png
- Directory: /path/to/dir/

Therefore, a path is valid, if:
- It begins with one forward slash
- Levels are represented by one forward slash (no backslashes allowed)

I've taken a look over the static methods in `Path` but haven't found a method that meets these requirements since microsoft likes to use backslashes in their paths.

Reply:
Okay. Thanks for elaborating!